### PR TITLE
Added password change UI

### LINKED ISSUE #827: Account Settings: Change Password & Notification Settings
See https://invis.io/WZLBNGQ5CKY

### acceptance criteria
- [ ] Add tab interface to (account) settings page
- [ ] Put current notification settings under "Email Notifications" tab (keep existing styling). Note that this form is also accessed by users that are not logged in and we should preserve that behavior 
- [ ] Add "Account" tab that displays all social auths . 
- [ ] Click to change password
- [ ] Users who do not have a email social auth should not be allowed to change their password
- [ ] Change password form (see https://projects.invisionapp.com/share/WZLBNGQ5CKY#/screens/303520669)

### out of scope
- "remove" account 
- change e-mail address
- styling update

### PR DESCRIPTION
**Pull request by gsidebo**
#### What are the relevant tickets?
Closes #827 

#### What's this PR do?
Adds the password change UI

#### How should this be manually tested?
With an email-authenticated user, go to the settings page, click the "Account" tab, click "Change Password", then use the form to change your password

#### Where should the reviewer start?
Probably `static/js/containers/AccountSettingsPage.js` and `static/js/containers/PasswordChangePage.js`

#### Any background context you want to provide?
The designs for the account settings page are incomplete. The output for Micromasters-authenticated users is just a placeholder for now.

#### Screenshots

ss 2018-07-11 at 10 42 48
ss 2018-07-11 at 10 42 59
ss 2018-07-11 at 10 47 45

### CODE REVIEW

**Comment by gsidebo** [comment]
Marking this WIP for the following reasons:

1. After the password change form successfully submits, a message is shown on the same page, but subsequent visits to the form page still show the success message instead of showing the form again (if there was no browser refresh). We don't seem to have a clear pattern for this type of UX and the options for a direct solution don't seem very straightforward. We might want to change the approach
1. If you partially fill out the password change form, then use the Cancel link button, chrome asks if you want to save the password even though you didn't submit the form. I have some ideas for tackling that but wanted to get a PR up before coming up with a possibly-finicky solution to that minor bug
1. Still have tests + linting to do

**Comment by pdpinch** [comment]
When you get a chance, can you add some screenshots to the description?

**Comment by gsidebo** [comment]
Yup, was planning on doing that this AM. Coming right up...

**Comment by gsidebo** [comment]
@pdpinch Screenshots added. The first one is the most important to review. A few decisions we need to make:

1. If we want to add the logos as part of this PR
1. If the answer to (1) is yes, what logo to use for Micromasters
1. What exactly to show for the Micromasters auth line. I had this showing the user email earlier, and I can add that back in. I was wondering if that would be confusing since the user didn't actually use that email to log into OD, and if they authenticate via email as well, we would be showing the email twice. That seems like it wouldn't be too jarring but thought it was worth flagging

**Comment by gsidebo** [comment]
@pdpinch One other thing to consider: long email addresses on the account settings page. A long email would bump the "Change Password" link to the next line. One way to handle this would be to have the end of the email address "fade" at the end before bumping into the link. This could be accomplished with some HTML/CSS tweaks. Might be worth asking Rob/Lucas for their opinions

**Comment by pdpinch** [comment]
We can do the logos later. 

For the long e-mail address, are you just talking about making the div wider?

**Comment by gsidebo** [comment]
We won't want an extra-wide div for narrow browser widths (mobile in particular). I was suggesting that everything could stay on one line, but the "Change Password" link would basically hide the email address underneath with a fading effect if the email was long enough. Something like [this](https://codepen.io/martinwolf/pen/DkKgJ) (just imagine the "Change Password" link to the right of that faded portion)

File: authentication/views.py [hunk 1]
```diff
@@ -137,41 +169,51 @@ def send(self, to, *args, **kwargs):
...
-            return Response({}, status=status.HTTP_200_OK)
+        Overrides CustomDjoserAPIView._action. If the settings indicate that the user
+        should not be logged out after a successful password change, the session is
+        updated. Without this explicit refresh, the user's session would be invalid
+        and they would be logged out.
+        """
+        response = super()._action(serializer)
+        if (
+                response.status_code in (status.HTTP_200_OK, status.HTTP_204_NO_CONTENT) and
+                settings.DJOSER['LOGOUT_ON_PASSWORD_CHANGE'] is False
```
Comment (mbertrand) [inline authentication/views.py]: Would this setting ever be `True`?  If not, maybe remove the check for it here?

Comment (mbertrand) [reply authentication/views.py]: And remove the setting altogether?

Comment (gsidebo) [reply authentication/views.py]: I included this just to guard against a future case where we change this setting, but looking at it now I agree with you. I'll remove this logic. I'll keep the setting just to be explicit about the behavior we want

**Comment by gsidebo** [comment]
My bad. Fixed

**Pull request merged**